### DISCUSSION TITLE
FS25 Courseplay ?

### DISCUSSION THREAD

**Discussion by bojanh66** [Courseplay/Courseplay_FS22#3328] [General]
I am curious if you guys will bring CP to FS25 ? Your mod is one of the most important mod for single player gamers managing big farms, playing like farm manager, along with Autodrive.

**Comment by Tensuko** [comment]
Planed, but let us wait untill we know more important stuff about FS25 to tell more about how we do it.

**Comment by bojanh66** [comment]
https://www.instagram.com/p/C9FG_bJMSoD/?img_index=1

It seems AI worker in Fs25 will use some kind of Courseplay ???

**Reply by Tensuko** [reply]
Relax... Different System and sure not that many options.
As said, let us get more Informations first, I am on it collecting informations and we will see what and how we do it.

So don't panic :)

**Reply by coatsy35** [reply]
Exciting though. Could be many more good things come from what they’ve had to do themselves and allow you even more functionality. 😎

**Comment by VergylTantor** [comment]
I'm very relived to hear that you all are planning to bring Courseplay to FS25. Was waiting to preorder until I had some answers to a few questions. Thank you for answering one question and removing one barrier to my preorder.
🎈 🎉🥳 🎉🎈

**Comment by Mnm974** [comment]
Hi! I know the game literally just came out. So I was just wondering when course play would be available? The current AI in the game is okay but not as great as what you guys have given us in FS22

**Reply by Tensuko** [reply]
We don't name any date. We don't know it either.

**Reply by Mnm974** [reply]
That’s fair, I’ll keep myself updated on whenever it’ll come. Looking forward to it coming into the game

**Reply by pvaiko** [reply]
Giants doesn't really make this easy for us, as usual, they changed their APIs but don't yet have the scripts (or at least the documentation) available, also no FS25 debugger, once we have those, we can make progress. I'm too old and frustrated to reverse engineer their stuff :)

**Comment by Redster999** [comment]
For me I think the release was good and the ingame socalled "courseplay" is ok. I dont expect you guys to rush anything as it is probably a lot of work to make this from scratch. But the AI driver is much better now, (point a to b) not at all perfect but still much better. 
For me the lacking parts are baling, handling the bales on the fields and unloading combines. When it comes to AI silagework and shoveling you guys did an awesome work with fs22. My take for what it is worth is perhaps going back to how you guys did it with fs19. 
Again, Thanks for the work you guys are doing.

**Reply by Tensuko** [reply]
There won't be anything like in FS19.
Always forwards, never backwards.

**Reply by Mnm974** [reply]
I love this reply. Always forwards, never backwards. When I saw FS25 and their AI I immediately thought to myself that once the Courseplay devs implement it, it’s probably going to be crazy!

**Reply by Mnm974** [reply]
As for the ai in the game now, unloading combines is a big one for me and the fact I still have to do it manually for now sucks

**Reply by Redster999** [reply]
Forwards not backwards. Nice. Really looking forward to whatever ideas you guys are going to do.

**Comment by k4talin** [comment]
Sadly . The well advertised integrated AI is a sad joke. Almost zero customizations. It is so Generic/Minimal.It is "a bit better than nothing" :))))

**Reply by Tensuko** [reply]
I didn't expect more then that.

**Reply by Mnm974** [reply]
It's honestly very underwhelming, very very barebones as compared to the work you all put in FS22 so I'm very excited to see what the FS25 version has in store

**Reply by MoslessNeo** [reply]
For FS25, will we have a dedication for the Case IH Austoft 8800 Multi-Row sugarcane harvester and the auger wagon?
It's been since FS15 that we haven't had any attention from the development team for this part of the game.
The CP has evolved a lot and I sincerely hope that you can make it work in this version of the game.

**Reply by Tensuko** [reply]
Not a priority...
Just a very few ppl use this and we have to make all the other stuff working first.

We will see.

**Reply by k4talin** [reply]
> I didn't expect more then that.

I’m shocked to see a few people here saying the in-game AI is "okay" or "usable." It’s garbage. It does nothing—no resume, no auto unload combines, no multitool support, as well as many many many settings/customisations which Courseplay provide and AI does not. It’s only somewhat functional for small fields. Question for guys said ingame AI is ok for now: Are you seriously going to use ONE TOOL on a 6HA or larger field and wait 4 hours or more to plow or harvest slow-growing crops like potatoes,parsnip and all slow harvesting crops they introduced?

Please stop calling the in-game AI "Courseplay." It’s not even 10% of what Courseplay offers. This is what a team of professionals came up with? Garbage. And the lane assist—OMG. On top of all that, they didn’t even release the Giants Editor.

After seeing what Giants did with their AI, I’m even more impressed by the level of expertise demonstrated by @Tensuko and his team.
Much Love @Tensuko

**Reply by tomtenberge** [reply]
k4talin i think they are comparing it to the AI in fs22, not with courseplay. compared to FS22 the ai IS okay and usable, compared to courseplay, it is garbage.

**Reply by k4talin** [reply]
> k4talin i think they are comparing it to the AI in fs22, not with courseplay. compared to FS22 the ai IS okay and usable, compared to courseplay, it is garbage.

I do agree. Compared to FS22 it is huge improvement. At least it does finish jobs.

**Reply by Mnm974** [reply]
Well of course compared to courseplay its trash, but yes comparing the current ai to the in-game ai of FS22 this one is better but not by much, it's able to do the bare minimum but none of the advanced work. Since launch I've been avoiding doing large fields just because of how long it'll take without multi-tools

**Reply by k4talin** [reply]
Same here. I did some gimmics to workaround on large fields with 2 tools. Like set up max number of headlands and put one tool start on headlands while the other will start on rows. That will do some how. If you intrerupt them OMG, chaos starting ;)

**Reply by coatsy35** [reply]
Have you seen this video? I hope that mod makers will be able to tap into their line generation system as it's so quick and adapts to the fields. Really nice to see. Whether or not the other features CP has like round corners and lands, racetrack etc... can utilise this system remain to be seen I guess.

https://www.youtube.com/watch?v=tAKzvkUyVUg&ab_channel=TheFarmSimGuy

**Reply by Tensuko** [reply]
Will definitly not use it.
- no waypoint system
- nothing there is round
- it is only fast because of simple shapes and maybe not in lua

**Reply by pvaiko** [reply]
Not sure what you mean by "no waypoint system", there'll like waypoints they just don't show them. 

Since we don't know anything about it yet, I'd refrain from strong statements :)

**Reply by coatsy35** [reply]
I’m hoping they have it simple for cross platform and that modders will be able to tap into it with more granularity for PC and enhance it for curves etc… surely they will give access to it in LUA. Even in new FS25 ebook they have they talk about CP and Wopsters guidance steering as examples of enhanced steering mods

**Comment by TheWhiteRussian** [comment]
Just donated some money, came to say I really appreciate all the work that has been done with courseplay. I only bought this game because of the planned implementation of this mod. Much love from Canada.

**Reply by pvaiko** [reply]
Thank you, what motivates us most is positive feedback, donations are never expected but always appreciated.

**Reply by k4talin** [reply]
@TheWhiteRussian God bless you brother.

**Comment by tomtenberge** [comment]
Yeah, the build in systen leaves much to be desired

Op vr 15 nov 2024 15:40 schreef Tensuko ***@***.***>:

**Comment by JarnoDeBoerNL** [comment]
hey guys n gals do we have any info on this mod? i really REALLY dont like this ingame courseplay im really missing some key importand features such as being able to safe the course

**Reply by Tensuko** [reply]
Well, there is no ingame courseplay, its a Giants Helper.
As you can read, there is not much Info yet, you need to have patient untill we have it rdy wich will take longer as there was to many stuff changed. As soon as I have something that is worth showing, I will show it.

**Reply by k4talin** [reply]
> Well, there is no ingame courseplay, its a Giants Helper. As you can read, there is not much Info yet, you need to have patient untill we have it rdy wich will take longer as there was to many stuff changed. As soon as I have something that is worth showing, I will show it.

Hell Yea Brother. Hell Yea.

**Reply by k4talin** [reply]
Imagine. No course play. No autoload. No Autodrive. Life sucks :)

**Reply by k4talin** [reply]
And they ruined missions as well. Most of them worthless.

**Reply by BoxerHUN** [reply]
By the time I accepted FS22, I learned to use CP and AD, by then FS25 came out... Now it's all starting over. :D Currently FS22 is more usable than FS25. The creation of custom routes is still missing.

**Reply by JarnoDeBoerNL** [reply]
> Well, there is no ingame courseplay, its a Giants Helper. As you can read, there is not much Info yet, you need to have patient untill we have it rdy wich will take longer as there was to many stuff changed. As soon as I have something that is worth showing, I will show it.

thank you very much

**Reply by JarnoDeBoerNL** [reply]
> And they ruined missions as well. Most of them worthless.

oh yeahhhh i hate the timer aswell

**Reply by JarnoDeBoerNL** [reply]
> By the time I accepted FS22, I learned to use CP and AD, by then FS25 came out... Now it's all starting over. :D Currently FS22 is more usable than FS25. The creation of custom routes is still missing.

i understand that struggle luckily for me courseplay is the most used mod next to easy development controls (i hope that mod comes to fs25 too)  for me

**Reply by Tensuko** [reply]
I actually like the new missions. More different tasks.
The money gained could be adjusted with mods, so I would not worry too much about it.

**Reply by JarnoDeBoerNL** [reply]
> I actually like the new missions. More different tasks. The money gained could be adjusted with mods, so I would not worry too much about it.

yeah thats fine but i dont like the timer u get

**Reply by antonkapralov** [reply]
> By the time I accepted FS22, I learned to use CP and AD, by then FS25 came out... Now it's all starting over. :D Currently FS22 is more usable than FS25. The creation of custom routes is still missing.

Without Curseplay and Autodrive FS25 is not playeble and / or enjoyble to play. i manage to load the curse play mod in FS25 but not working, and autodrive to, i am getting keybinding in FS25 for autodrive setting, but not working (activating them has no result). Has log with errors, dont know if you can use this errors, (lots of erros like this -> Gui profile 'xxxxxxxx' still uses old format, please convert it to the new one, how do you conver it ? their is still no editor 10 from giant and in 9 you cant ( at list i dint find the way how to).

cant wait till your mod will work again. I am getting more plesure from FS22 right now.

**Comment by dmb25108** [comment]
Just so you are aware there are websites claiming to have courseplay fs25
https://fs25planet.com/tutorials-cheats/courseplay-for-fs25-detailed-review.html

**Reply by Tensuko** [reply]
All fake and stolen shit.

**Reply by antonkapralov** [reply]
right now it is easy to do as ppl looking for most wanted mod like courseplay and someone can abuse it.

**Reply by BrutalRIP** [reply]
this always happens unfortunately

**Reply by pull2play** [reply]
Happens every year. People are desperate for Courseplay and Autodrive as you simply cannot maintain a large(-ish) farm without them. Simply indispensable. I always tell myself to wait until a new Courseplay version hits beta before buying the next FS, but yet again I find myself swearing at the useless in-game AI, this time in FS25. You'd think I'd have learned my lesson after 15 years.

tl;dr If it isn't available here, it's fake and almost certainly doesn't work. Don't bother.

**Reply by BoxerHUN** [reply]
Unfortunately, it spreads like wildfire. :(

**Reply by antonkapralov** [reply]
https://farmingsimulator25mods.com/courseplay-beta-v1-0/, dont know what is point to place this one as it is <version>7.5.0.1</version>, and not working at all.

**Reply by k4talin** [reply]
> https://farmingsimulator25mods.com/courseplay-beta-v1-0/, dont know what is point to place this one as it is 7.5.0.1, and not working at all.

Saw that too. That page is Lame. Starting with title and Download button.

**Comment by tjwoten79** [comment]
Hey, thanks for all you do!!  Courseplay makes Farming Simulator 100x better.  And Autodrive is #2 in the necessities list.  I can't wait for the next version to come out!  I wish I was extremely rich so I could throw thousands in your direction, however since that's impossible I can give you a thank you!!  Now that it thumps my brain though, is there a way around here to donate to the cause?  Thanks again.

**Reply by Redster999** [reply]
I bet they love the comments still ;)
Yes there is a way. Scroll down to the bottom 
https://github.com/Courseplay/Courseplay_FS22

**Reply by Darrenishere** [reply]
I agree 100% FS without CoursePlay is like a burger without a bun. FS25 without Autodrive is like a burger without the burger. Just the ability to have a couple combines in a large field, chaser bins and a truck/s delivering grain to a silo is just so fulfilling. I know no everyone plays like this, but for those that do, my main game is still on FS22, and likely will be for awhile yet.

**Reply by fudoido** [reply]
That is the way i usually play FS, i find myself spending more time making AD routes and saving CP courses than playing the actual game lol.
I think of it like i am the farm owner, need to manage, so i gotta hire staff and make them work. make use of AD and CP for that and i can do other stuff none of the mods can do, like wood cutting.

**Comment by Boandlgramer** [comment]
Bis wann kann man ungfefähr mit Autodrive und Courseplay für LS25 rechnen. Ohne die beiden ist das Spiel für mich nicht spielbar.
KI kann man total vergessen. Hoffe ihr bekommt das in der nächsten Zeit zum laufen. Angeblich gibt es schon eine Betaversion. Was man da so sieht, wäre das ja schon in Ordnung. Lieber eine nur halbwegs fertige Beta als gar keine.

**Reply by pvaiko** [reply]
We are almost there...

**Reply by Tensuko** [reply]
Aufgrund gewisser Umstände war ich mehr oder weniger gezwungen eine Beta an ein paar von mir bekannte YouTuber zu verteilen (Thema Gleichberechtigung, es ist kompliziert...).
Hat für uns alle gerade den Vorteil, dass wir noch ein paar bugs finden konnten und diese beseitigen können.
Habt noch etwas geduld, wir sind bald soweit.

"Lieber eine nur halbwegs fertige Beta als gar keine." 

Das sagt ihr jetzt so, aber wenn es nur halb da ist, dinge die nicht gehen da sind, Spiel nicht speichert etc wird es genug Leute geben die sich beschweren. Du vlt nicht, aber genug andere.

**Reply by fudoido** [reply]
agree, a "half finished beta" made public would just flood the github with bug reports and people complaining all over the internet on "how bad it is" when they dont realize it is still a beta..... at this stage i would even consider it an alpha version, as many changes are still being worked on.

**Reply by King-Ranch** [reply]
Hello,
I created an account just to comment on this project.   I want to sincerely thank you for all of your work on this.  I can't imagine the time it takes.   It is a game-changer for Farming Simulator.  

Please take time now to enjoy your families during the holidays.   We will be patiently waiting for a release in the new year when you feel a release is ready.   While we very much appreciate your updates, you certainly don't owe any of us reasons for why and when you do things.    Thank you again and Merry Christmas.

**Reply by CrazyCarrot911** [reply]
Hallo DevTeam,

Vielen Dank für die Mühe das in LS25 reinzubasteln ;)

Ohne Euch ist meine Mais Saison in Gefahr..O_O   deshalb noch schnell die 3 Kaffee ;)


Ihr seit einfach 1. Klasse !

Bitmaster69
CrazyCarrot911

**Comment by Tensuko** [comment]
Please read the changelog! :)
https://github.com/Courseplay/Courseplay_FS25/releases

**Comment by CrazyCarrot911** [comment]
Danke ;)

Mein erster Eindruck, Mais ernten geht besser als vorher. Das Wenden beider Kontrahenten geht viel flüssigern als vorher.

Top !

**Comment by nitroSkirata** [comment]
Where do you guys want the bug reports to be filed?
Currently Issues aren't available in the FS25 fork

PS: Awesome work, can't wait to see where this is all headed in the coming months

**Reply by Tensuko** [reply]
Issues available soon

**Comment by Canyon202202** [comment]
Hello friends, please explain how to install coursplay. I put mods in the folder, but nothing happens. I would be grateful for the information.

The game itself sees the courseplay, but when you go into mods to select a map, it is not displayed there. I wanted to create a new career, but there are no courses to choose from.

For communication: monolim_28@icloud.com

**Reply by Tensuko** [reply]
Like all other mods. Download and put in mods folder.

https://github.com/Courseplay/Courseplay_FS25/releases

**Reply by Behnert18** [reply]
same here

I have it in the right folder. it is also displayed in the game under mods, but it is always missing when starting a new game

**Comment by superhuff20** [comment]
Once you have a saved game, and update the mod.  How do I get courseplay to work again?  It shows up in my mods but cannot use in my save game.  Thanks!

**Reply by Behnert18** [reply]
I have the same problem!
In the mod folder and game started. is displayed under mods but when I start a new game dei mod is missing!

is there a solution in the meantime? do more people have this problem?

**Comment by CigarD313** [comment]
I'm trying to save my courses and when I hit Create Folder, it's asking for a password.  Any help on this?

**Reply by Tensuko** [reply]
????????

**Reply by CigarD313** [reply]
![2300320_28](https://github.com/user-attachments/assets/f8a25ea9-6883-4ad8-859f-1c1780279881)

**Reply by CigarD313** [reply]
I manually added the F3 folder and Harvester subfolder.  Still would not let me save it, this is the screen I'm getting.

**Reply by Danneman79** [reply]
I have the same problem on FS25_Papenburger_Map_v2_4k_V2_2_9a, if you type in some numbers and enter, it adds to my money.
It works fine on other maps with the same mods.

**Reply by smigly** [reply]
Same here. I have manually created the folder and it did let me save courses for a while. After some time playing it asks for password. After restart it works again for some time and then it asks for password again. I have not notice what triggers it and every time it is a random amount of time. I am also using autodrive, so maybe there is something connected to AD?
I playing on vanilla map.

**Comment by CigarD313** [comment]
Once you start the map,  save it,  quit and restart the map,  you should be able to save your courses.Sent from my Galaxy
-------- Original message --------From: Behnert18 ***@***.***> Date: 1/20/25  9:48 AM  (GMT-05:00) To: Courseplay/Courseplay_FS22 ***@***.***> Cc: CigarD313 ***@***.***>, Comment ***@***.***> Subject: Re: [Courseplay/Courseplay_FS22] FS25 Courseplay ? (Discussion #3328) 
same here
I have it in the right folder. it is also displayed in the game under mods, but it is always missing when starting a new game

**Reply by Behnert18** [reply]
that doesn't solve the problem. The mod is still not displayed for selection at the start of the game
